.drone-v1.yml

```diff
@@ -0,0 +1,252 @@
+  image: quay.io/ukhomeofficedigital/kd:v1.16.0
+  commands:
+    - cd kube-hocs-templates
+    - ./deploy.sh
+  environment:
+    ENVIRONMENT: cs-dev
+    HOCS_TEMPLATES_CS_DEV:
+      from_secret: hocs_templates_cs_dev
+    KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+    VERSION: build-${DRONE_BUILD_NUMBER}
+  when:
+    branch:
+      - master
+    event:
+      - push
+  depends_on:
+    - Clone Kube Repo
+
+- name: Deploy To WCS-DEV
+  image: quay.io/ukhomeofficedigital/kd:v1.16.0
+  commands:
+    - cd kube-hocs-templates
+    - ./deploy.sh
+  environment:
+    ENVIRONMENT: wcs-dev
+    HOCS_TEMPLATES_WCS_DEV:
+      from_secret: hocs_templates_wcs_dev
+    KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+    VERSION: build-${DRONE_BUILD_NUMBER}
+  when:
+    branch:
+      - master
+    event:
+      - push
+  depends_on:
+    - Clone Kube Repo
+
+- name: Generate & Tag Build
+  image: quay.io/ukhomeofficedigital/hocs-version-bot:latest
+  commands:
+    - >
+      /app/hocs-deploy
+      --dockerRepository=quay.io/ukhomeofficedigital
+      --environment=qa
+      --registryPassword=$${DOCKER_PASSWORD}
+      --registryUser=ukhomeofficedigital+hocs_quay_robot
+      --service=${SERVICE}
+      --serviceGitToken=$${GITHUB_TOKEN}
+      --sourceBuild=$${IMAGE_VERSION}
+      --version=$${SEMVER}
+      --versionRepo="https://gitlab.digital.homeoffice.gov.uk/hocs/hocs-versions.git"
+      --versionRepoServiceToken=$${GITLAB_TOKEN}
+  environment:
+    DOCKER_PASSWORD:
+      from_secret: QUAY_ROBOT_PASSWORD
+    GITLAB_TOKEN:
+      from_secret: GITLAB_TOKEN
+    GITHUB_TOKEN:
+      from_secret: GITHUB_TOKEN
+  when:
+    event:
+      - promote
+    target:
+      - release
+
+- name: Deploy to CS-QA
+  image: quay.io/ukhomeofficedigital/kd:v1.16.0
+  commands:
+    - source version.txt
+    - echo $VERSION
+    - cd kube-hocs-templates
+    - ./deploy.sh
+  environment:
+    ENVIRONMENT: cs-qa
+    HOCS_TEMPLATES_CS_QA:
+      from_secret: hocs_templates_cs_qa
+    KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+  when:
+    event:
+      - promote
+    target:
+      - release
+  depends_on:
+    - Clone Kube Repo
+    - Generate & Tag Build
+
+- name: Deploy to WCS-QA
+  image: quay.io/ukhomeofficedigital/kd:v1.16.0
+  commands:
+    - source version.txt
+    - echo $VERSION
+    - cd kube-hocs-templates
+    - ./deploy.sh
+  environment:
+    ENVIRONMENT: wcs-qa
+    HOCS_TEMPLATES_WCS_QA:
+      from_secret: hocs_templates_wcs_qa
+    KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+  when:
+    event:
+      - promote
+    target:
+      - release
+  depends_on:
+    - Clone Kube Repo
+    - Generate & Tag Build
+
+- name: Deploy to Not Prod 
+  image: quay.io/ukhomeofficedigital/kd:v1.16.0
+  commands:
+    - cd kube-hocs-templates
+    - ./deploy.sh
+  environment:
+    ENVIRONMENT: ${DRONE_DEPLOY_TO}
+    HOCS_TEMPLATES_CS_DEV:
```

Comment:
I wonder if `HOCS_TEMPLATES_${DRONE_DEPLOY_TO/-/_/}` would work here? Would save us having to update every drone file every time we get a new environment set up. I'll ask if we can use per-service tokens instead of per-namespace ones -- I think that'd cause considerably less faff now we're dealing with many namespaces.

Reply:
Just had a play - doesn't look like you can do that here.